R-package/configure

```diff
@@ -0,0 +1,3017 @@
+#! /bin/sh
```

Comment:
as the text here says, this script was automatically generated by `autoconf`. The command used to generate it can be seen in `build-cran-package.sh`.

Reviewers should not take time reviewing its contents (3000 lines of hard-to-read shell code), and we should never edit this file by hand.

Reply:
I think we can add a check at CI to make sure that contributors don't forget to re-generate this file after modifying configuration. Refer to 
https://github.com/microsoft/LightGBM/blob/cfc5e4fe8b2dddc2c3794267d461246ae8cddc35/.ci/test.sh#L34-L39

BTW, the same can be done for `Rd` files (in a separate PR, of course), I guess...

Reply:
oh interesting! Ok I like that idea a lot. That is better than having to "just remember"

Reply:
I just added #3205 for the `Rd` files. Will add similar logic here for `configure`